### ISSUE TITLE
Auth problem

### ISSUE THREAD

**Issue by 53rg0** [bryantgillespie/nuxt3-directus-starter#6]
When i already logged in after a while i get error in NUXT client:

  ```
`Token found, fetching user from server
Exception in PromiseRejectCallback:
@directus/sdk:819
}
^

RangeError: Maximum call stack size exceeded
Exception in PromiseRejectCallback:
@directus/sdk:923
await this._auth.refreshIfExpired();
^

RangeError: Maximum call stack size exceeded
Token is eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjU0ZmRlYmJjLTdkZTYtNGY5Ni05N2ExLWZhNTdkNzk3YzVlNCIsInJvbGUiOiI0YTE2NWVlNS0wZDgzLTRjYjQtOGYwZC1hMjBmMGNkY2RjZjciLCJhcHBfYWNjZXNzIjoxLCJhZG1pbl9hY2Nlc3MiOjEsImlhdCI6MTY2MjkwMTU4NCwiZXhwIjoxNjYyOTAyNDg0LCJpc3MiOiJkaXJlY3R1cyJ9.LcCnF_3xGtR4KVbc6lAJFe7_eU-pKyRjHHGu_GzY4Cg
TransportError: Maximum call stack size exceeded
at Transport.request (@directus/sdk:700:19)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Transport.get (@directus/sdk:704:16)
at async MeHandler.read (@directus/sdk:429:26)
at async Proxy.getUser (/Users/53rg0/Sites/TEST/store/auth.ts:56:22)
at async /Users/53rg0/Sites/TEST/plugins/directus.ts:39:85
at async Object.callAsync (file:///Users/53rg0/Sites/TEST/node_modules/unctx/dist/index.mjs:53:16)
at async applyPlugin (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/nuxt.mjs:74:23)
at async Module.applyPlugins (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/nuxt.mjs:84:5)
at async createNuxtAppServer (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/entry.mjs:30:7) {
  parent: TransportError: Maximum call stack size exceeded
  at Transport.request (@directus/sdk:700:19)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async Transport.post (@directus/sdk:719:16)
  at async refreshPromise (@directus/sdk:825:30)
  at async Auth.refreshIfExpired (@directus/sdk:814:13)
  at async Transport.beforeRequest (@directus/sdk:923:21)
  at async Transport.request (@directus/sdk:668:22)
  at async Transport.get (@directus/sdk:704:16)
  at async MeHandler.read (@directus/sdk:429:26)
  at async Proxy.getUser (/Users/53rg0/Sites/TEST/store/auth.ts:56:22) {
    parent: TransportError: Maximum call stack size exceeded
    at Transport.request (@directus/sdk:700:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Transport.post (@directus/sdk:719:16)
    at async refreshPromise (@directus/sdk:825:30)
    at async Auth.refreshIfExpired (@directus/sdk:814:13)
    at async Transport.beforeRequest (@directus/sdk:923:21)
    at async Transport.request (@directus/sdk:668:22)
    at async Transport.post (@directus/sdk:719:16)
    at async refreshPromise (@directus/sdk:825:30)
    at async Auth.refreshIfExpired (@directus/sdk:814:13) {
      parent: [TransportError],
        response: undefined,
        errors: []
    },
    response: undefined,
      errors: []
  },
  response: undefined,
    errors: []
}

ERROR  [h3] [unhandled] Parse Error: Header overflow                                                                                                             16:35:08

at Socket.socketOnData (node:_http_client:521:22)
at Socket.emit (node:events:513:28)
at Socket.emit (node:domain:489:12)
at addChunk (node:internal/streams/readable:324:12)
at readableAddChunk (node:internal/streams/readable:297:9)
at Readable.push (node:internal/streams/readable:234:10)
at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)

Token found, fetching user from server
Exception in PromiseRejectCallback:
@directus/sdk:819
}
^

RangeError: Maximum call stack size exceeded
Exception in PromiseRejectCallback:
@directus/sdk:923
await this._auth.refreshIfExpired();
^

RangeError: Maximum call stack size exceeded
Token is eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjU0ZmRlYmJjLTdkZTYtNGY5Ni05N2ExLWZhNTdkNzk3YzVlNCIsInJvbGUiOiI0YTE2NWVlNS0wZDgzLTRjYjQtOGYwZC1hMjBmMGNkY2RjZjciLCJhcHBfYWNjZXNzIjoxLCJhZG1pbl9hY2Nlc3MiOjEsImlhdCI6MTY2MjkwMTU4NCwiZXhwIjoxNjYyOTAyNDg0LCJpc3MiOiJkaXJlY3R1cyJ9.LcCnF_3xGtR4KVbc6lAJFe7_eU-pKyRjHHGu_GzY4Cg
TransportError: Maximum call stack size exceeded
at Transport.request (@directus/sdk:700:19)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Transport.get (@directus/sdk:704:16)
at async MeHandler.read (@directus/sdk:429:26)
at async Proxy.getUser (/Users/53rg0/Sites/TEST/store/auth.ts:56:22)
at async /Users/53rg0/Sites/TEST/plugins/directus.ts:39:85
at async Object.callAsync (file:///Users/53rg0/Sites/TEST/node_modules/unctx/dist/index.mjs:53:16)
at async applyPlugin (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/nuxt.mjs:74:23)
at async Module.applyPlugins (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/nuxt.mjs:84:5)
at async createNuxtAppServer (/Users/53rg0/Sites/TEST/node_modules/nuxt/dist/app/entry.mjs:30:7) {
  parent: TransportError: Maximum call stack size exceeded
  at Transport.request (@directus/sdk:700:19)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async Transport.post (@directus/sdk:719:16)
  at async refreshPromise (@directus/sdk:825:30)
  at async Auth.refreshIfExpired (@directus/sdk:814:13)
  at async Transport.beforeRequest (@directus/sdk:923:21)
  at async Transport.request (@directus/sdk:668:22)
  at async Transport.get (@directus/sdk:704:16)
  at async MeHandler.read (@directus/sdk:429:26)
  at async Proxy.getUser (/Users/53rg0/Sites/TEST/store/auth.ts:56:22) {
    parent: TransportError: Maximum call stack size exceeded
    at Transport.request (@directus/sdk:700:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Transport.post (@directus/sdk:719:16)
    at async refreshPromise (@directus/sdk:825:30)
    at async Auth.refreshIfExpired (@directus/sdk:814:13)
    at async Transport.beforeRequest (@directus/sdk:923:21)
    at async Transport.request (@directus/sdk:668:22)
    at async Transport.post (@directus/sdk:719:16)
    at async refreshPromise (@directus/sdk:825:30)
    at async Auth.refreshIfExpired (@directus/sdk:814:13) {
      parent: [TransportError],
        response: undefined,
        errors: []
    },
    response: undefined,
      errors: []
  },
  response: undefined,
    errors: []
}

ERROR  [h3] [unhandled] Parse Error: Header overflow                                                                                                             16:35:18

at Socket.socketOnData (node:_http_client:521:22)
at Socket.emit (node:events:513:28)
at Socket.emit (node:domain:489:12)
at addChunk (node:internal/streams/readable:324:12)
at readableAddChunk (node:internal/streams/readable:297:9)
at Readable.push (node:internal/streams/readable:234:10)
at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
`
```

**Comment by 53rg0** [comment]
Seems something with

const user = await $directus.users.me.read({
          fields: ['*'],
        })

in /store/auth.ts

**Comment by 53rg0** [comment]
maybe (i am not sure) token expired but it try to get user data with old token, not refresh token

any way, error when running getUser

```
`async getUser() {
      const { $directus } = useNuxtApp()
      try {
        // Try to fetch the user data
        const user = await $directus.users.me.read({. 
          fields: ['*'],                                                        
        })                                                                           
        // Update the auth store with the user data
        this.loggedIn = true
        this.user = user
      } catch (e) {
        console.log(e)
        // await helpers.clear()
      }
    },
````

**Comment by 53rg0** [comment]
Seems like something with directus auth.
Error appears after around 15 min, and it is the same time as directus session life time

# The duration that the access token is valid ["15m"]
ACCESS_TOKEN_TTL="15m"

**Comment by azrikahar** [comment]
Let's close this favor of directus/sdk#53, just to keep all discussions in the same thread.

**Comment by azrikahar** [comment]
Linking the resolution here: https://github.com/directus/sdk/issues/53#issuecomment-1245152033

It should be okay to close this issue now 👍

**Comment by bryantgillespie** [comment]
Closed with 7a09cb42e10b7c5de8fb6dd83d69e68a181c0739
